coriolis/tests/cron/test_cron.py

```diff
@@ -0,0 +1,501 @@
+    def test_should_run_false(self, mock_is_expired, mock_compare):
+        mock_is_expired.return_value = True
+        dt = datetime.datetime.fromisoformat('2099-01-01')
+        self.cron._enabled = True
+        self.cron.schedule = {
+            'month': 1,
+            'hour': 0
+        }
+
+        self.assertRaises(
+            exception.CoriolisException,
+            self.cron.should_run,
+            None
+        )
+
+        result = self.cron.should_run(dt)
+
+        self.assertEqual(
+            False,
+            result
+        )
+
+        mock_is_expired.return_value = False
+        self.cron._enabled = False
+
+        result = self.cron.should_run(dt)
+
+        self.assertEqual(
+            False,
+            result
+        )
+
+        mock_compare.assert_not_called()
+
+    def test_send_status(self):
+        queue = mock.Mock()
+        self.cron._send_status(queue, mock.sentinel.status)
+        queue.put.assert_called_once_with(mock.sentinel.status)
+
+        queue.reset_mock()
+        self.cron._send_status(None, mock.sentinel.status)
+        queue.put.assert_not_called()
+
+    @mock.patch.object(cron.CronJob, '_send_status')
+    def test_start(self, mock_send_status):
+
+        self.cron.start()
+
+        self.cron._func.assert_called_once_with(
+            *self.cron._args, **self.cron._kw)
+        self.cron._on_success.assert_called_once_with(
+            self.cron._func.return_value)
+        mock_send_status.assert_called_once_with(
+            None,
+            {"result": self.cron._func.return_value,
+             "description": self.cron._description,
+             "name": self.cron.name,
+             "error_info": None}
+        )
+
+    @mock.patch.object(cron, 'LOG')
+    @mock.patch.object(sys, 'exc_info')
+    @mock.patch.object(cron.CronJob, '_send_status')
+    def test_start_on_error(
+        self,
+        mock_send_status,
+        mock_exc_info,
+        mock_LOG
+    ):
+        mock_exc_info.return_value = "mock_exc_info"
+        self.cron._func.side_effect = Exception('err_msg1')
+        self.cron._on_error.side_effect = Exception('err_msg2')
+
+        self.cron.start()
+
+        self.cron._func.assert_called_once_with(
+            *self.cron._args, **self.cron._kw)
+        self.cron._on_error.assert_called_once_with("mock_exc_info")
+        mock_send_status.assert_called_once_with(
+            None,
+            {"result": None,
+             "description": self.cron._description,
+             "name": self.cron.name,
+             "error_info": "mock_exc_info"}
+        )
+        mock_LOG.assert_has_calls([
+            mock.call.exception(self.cron._func.side_effect),
+            mock.call.exception(self.cron._on_error.side_effect)
+        ])
+
+
+@ddt.ddt
+class CronTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Cron."""
+
+    @mock.patch.object(schemas, 'validate_value')
+    def setUp(self, *_):
```

Comment:
What did you put the star for? 
```suggestion
    def setUp(self, _):
```